checker/checkInductive.ml

```diff
@@ -8,264 +8,155 @@
+      r1 r2
+  | (NotRecord | FakeRecord | PrimRecord _), _ -> false
+
+let check_inductive env mind mb =
+  let entry = to_entry mb in
+  let { mind_packets; mind_record; mind_finite; mind_ntypes; mind_hyps;
+        mind_nparams; mind_nparams_rec; mind_params_ctxt; mind_universes;
+        mind_private; mind_typing_flags; }
+    =
+    (** Locally set the oracle for further typechecking *)
+    let env = Environ.set_oracle env mb.mind_typing_flags.conv_oracle in
+    Indtypes.check_inductive env mind entry
+  in
+  let check = check mind in
 
-  (* Other propositions: elimination only to Prop *)
-  | InProp -> logical_sorts
+  Array.iter2 (check_packet env mind) mb.mind_packets mind_packets;
+  check "mind_record" (check_same_record mb.mind_record mind_record);
+  check "mind_finite" (mb.mind_finite == mind_finite);
+  check "mind_ntypes" Int.(equal mb.mind_ntypes mind_ntypes);
+  check "mind_hyps" (Context.Named.equal Constr.equal mb.mind_hyps mind_hyps);
+  check "mind_nparams" Int.(equal mb.mind_nparams mind_nparams);
 
-let check_predicativity env s small level =
-  match s, engagement env with
-      Type u, _ ->
-        (* let u' = fresh_local_univ () in *)
-        (* let cst = *)
-        (*   merge_constraints (enforce_leq u u' empty_constraint) *)
-        (*     (universes env) in *)
-        if not (UGraph.check_leq (universes env) level u) then
-          failwith "impredicative Type inductive type"
-    | Set, ImpredicativeSet -> ()
-    | Set, _ ->
-        if not small then failwith "impredicative Set inductive type"
-    | Prop,_ -> ()
+  check "mind_nparams_rec" (mb.mind_nparams_rec <= mind_nparams_rec);
+  (* module substitution can increase the real number of recursively
+     uniform parameters, so be tolerant and use [<=]. *)
 
-let sort_of_ind = function
-  | RegularArity mar -> mar.mind_sort
-  | TemplateArity par -> Type par.template_level
+  check "mind_params_ctxt" (Context.Rel.equal Constr.equal mb.mind_params_ctxt mind_params_ctxt);
+  ignore mind_universes; (* Indtypes did the necessary checking *)
+  ignore mind_private; (* passed through Indtypes *)
 
-let compute_elim_sorts env_ar params arity lc =
-  let inst = Context.Rel.to_extended_list Constr.mkRel 0 params in
-  let env_params = push_rel_context params env_ar in
-  let lc = Array.map
-    (fun c ->
-      Reduction.hnf_prod_applist env_params (Vars.lift (Context.Rel.length params) c) inst)
-    lc in
-  let s = sort_of_ind arity in
-  let infos = Array.map (Indtypes.infos_and_sort env_params) lc in
-  let (small,unit) = small_unit infos in
-  (* We accept recursive unit types... *)
-  (* compute the max of the sorts of the products of the constructor type *)
-  let min = if Array.length lc > 1 then Universe.type0 else Universe.type0m in
-  let level = Array.fold_left (fun max l -> Universe.sup max l) min infos in
-  check_predicativity env_ar s small level;
-  allowed_sorts small unit s
+  ignore mind_typing_flags;
+  (* TODO non oracle flags *)
```

Comment:
Do you plan to address this before merging?

Reply:
>Do you plan to address this before merging?

No.

Reply:
Were they already ignored before this PR?

Reply:
Yes (look around `let env0 =` and its uses before the PR)